grafana_client/client.py

```diff
@@ -188,3 +188,94 @@ def __request_runner(url, json=None, data=None, headers=None, accept_empty_json=
                     raise
 
         return __request_runner
+
+
+class AsyncGrafanaClient(GrafanaClient):
+    def __init__(
+        self,
+        auth,
+        host="localhost",
+        port=None,
+        url_path_prefix="",
+        protocol="http",
+        verify=True,
+        timeout=DEFAULT_TIMEOUT,
+        user_agent: str = None,
+        organization_id: int = None,
+    ):
+        super().__init__(
+            auth,
+            host=host,
+            port=port,
+            url_path_prefix=url_path_prefix,
+            protocol=protocol,
+            verify=verify,
+            timeout=timeout,
+            user_agent=user_agent,
+            organization_id=organization_id,
+        )
+        self.s = niquests.AsyncSession()
+        self.s.headers.setdefault("Connection", "keep-alive")
+
+    def __getattr__(self, item):
+        async def __request_runner(url, json=None, data=None, headers=None, accept_empty_json=False):
+            __url = "%s%s" % (self.url, url)
+            # Sanity checks.
+            if json is not None and not isinstance(json, (dict, list)):
+                raise TypeError(  # pragma: no cover
+                    f"JSON request payload has invalid shape. "
+                    f"Accepted are dictionaries and lists. "
+                    f"The type is: {type(json)}"
+                )
```

Comment:
:heavy_check_mark: done

little note aside, I didn't implement it as a class method, but rather as a util function.
because broke OOP principle of not using "self".

Reply:
Sure, can be static. Thanks!